src/services/login-util.js

```diff
@@ -0,0 +1,139 @@
+const http = require('http')
+const crypto = require('crypto')
+const consola = require('consola')
+const axios = require('axios')
+const config = require('./config')
+
+const AUTH0_DOMAIN = 'checkly'
+const AUTH0_CLIENT_ID = 'mBtwLFVm39GVZ1HpSRBSdRiLFucYxmMb'
+
+const generateAuthenticationUrl = (codeChallenge, scope, state) => {
+  const url = new URL(`https://${AUTH0_DOMAIN}.eu.auth0.com/authorize`)
+  const params = new URLSearchParams({
+    client_id: AUTH0_CLIENT_ID,
+    code_challenge: codeChallenge,
+    code_challenge_method: 'S256',
+    response_type: 'code',
+    redirect_uri: 'http://localhost:4242',
+    scope: scope,
+    state: state,
+  })
+
+  url.search = params
+  return url.toString()
+}
+
+function generatePKCE() {
+  const codeVerifier = crypto
+    .randomBytes(64)
+    .toString('base64')
+    .replace(/\+/g, '-')
+    .replace(/\//g, '_')
+    .replace(/=+$/, '')
+
+  const codeChallenge = crypto
+    .createHash('sha256')
+    .update(codeVerifier)
+    .digest('base64')
+    .replace(/\+/g, '-')
+    .replace(/\//g, '_')
+    .replace(/=+$/, '')
+
+  return {
+    codeChallenge,
+    codeVerifier,
+  }
+}
+
+function startServer(getToken) {
```

Comment:
I would rename the getToken to something more callbacky. Like, onTokenCallback

Reply:
Nice, dig it. Much clearer :+1: